caffeinated/admin/extend/registrations/EE_Event_Registrations_List_Table.class.php

```diff
@@ -100,17 +101,30 @@ protected function _set_properties()
                 $evt_id
             )
         ) {
-            $this->_bottom_buttons = array(
-                'report' => array(
-                    'route'         => 'registrations_report',
-                    'extra_request' =>
-                        array(
-                            'EVT_ID'     => $evt_id,
-                            'return_url' => $return_url,
-                            'incude_checkins' => true,
-                        ),
-                ),
-            );
+            if ($DTT_id) {
+                $this->_bottom_buttons = array(
+                    'report_datetime' => array(
+                        'route'         => 'registrations_report',
+                        'extra_request' =>
+                            array(
+                                'EVT_ID'     => $evt_id,
+                                'DTT_ID'     => $DTT_id,
+                                'return_url' => $return_url,
+                            ),
+                    ),
+                );
+            } else {
+                $this->_bottom_buttons = array(
+                    'report' => array(
+                        'route'         => 'registrations_report',
+                        'extra_request' =>
+                            array(
+                                'EVT_ID'     => $evt_id,
+                                'return_url' => $return_url,
+                            ),
+                    ),
+                );
+            }
```

Comment:
another potentially better way to handle this is to "isolate the thing that changes", which is a common quote from functional programming which could have also been written as "extract shared logic".

since there is a lot of duplication in the above code, we can separate things so that the shared logic is on its own, and then apply the changes in isolation. 

for example:

```php
$route_details = [
	'route'         => 'registrations_report',
	'extra_request' => [
		'EVT_ID'     => $evt_id,
		'return_url' => $return_url,
	]
];
if ($DTT_id) {
	$route_details['extra_request']['DTT_ID'] = $DTT_id;
	$this->_bottom_buttons['report_datetime'] = $route_details;
} else {
	$this->_bottom_buttons['report'] = $route_details;
}
```

Now I'm guessing that the `'report'` and `'report_datetime'` routes are very very similar, so you may want to think about consolidating those into a single route and then handle the differences based on the presence or absence of a datetime ID in the request parameters. But I'm totally guessing, and there may be bigger functional differences that require a separate route for those two reports. I'll totally leave it up to you to decide which path to take.

Reply:
You're right. I did it.